ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js

```diff
@@ -18,10 +17,24 @@ import {
   CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP,
   NETWORK_TO_NAME_MAP,
 } from '../../../../../shared/constants/network';
-import { getProviderConfig } from '../../../../ducks/metamask/metamask';
 
-export default function ConfirmationNetworkSwitch({ newNetwork }) {
-  const { chainId, nickname, type } = useSelector(getProviderConfig);
+const getNetworkDetails = (network) => {
+  if (network.nickname === undefined) {
+    delete network.nickname;
+  }
+  if (network.iconUrl === undefined) {
+    delete network.iconUrl;
+  }
```

Comment:
what do these deletes buy us?

Reply:
deleting the key makes the object spread on line 31 work properly. for example:
```
const a = { b: undefined };
const c = { b: '123', ...a };
console.log(c); // { b: undefined }
```

Reply:
ahh... would the following be an okay alternative? I feel that the inline makes it more obvious what is going on, which is assigning a default if not already defined.
```
  return {
    ...network,
    nickname: network.nickname ?? NETWORK_TO_NAME_MAP[network.chainId],
    iconUrl: network.iconUrl ?? CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[network.chainId],
  };
```

Reply:
yeah I could do that